code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +845,337 @@
+	if(expedite)
+		animate_hatch_king()
+		return
+	
+	marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP DETECTED IN [get_area_name(loc)].\n\nESTIMATED TIME UNTIL COMPLETION - ONE MINUTE.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+	var/datum/hive_status/hive
+	for(var/cur_hive_num in GLOB.hive_datum)
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("The King will hatch in approximately one minute."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King will hatch in approximately one minute."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+	timer = addtimer(CALLBACK(src, PROC_REF(animate_hatch_king)), 1 MINUTES, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+
+/// Causes the cocoon to change visually for hatching and initiates the next timer.
+/obj/effect/alien/resin/king_cocoon/proc/animate_hatch_king()
+	flick("hatching", src)
+	addtimer(CALLBACK(src, PROC_REF(hatch_king)), 2 SECONDS, TIMER_UNIQUE|TIMER_STOPPABLE)
+
+	marine_announcement("ALERT.\n\nEXTREME ENERGY INFLUX DETECTED IN [get_area_name(loc)].\n\nCAUTION IS ADVISED.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+	var/datum/hive_status/hive
+	for(var/cur_hive_num in GLOB.hive_datum)
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("All hail the King."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King has hatched!"), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+/// Actually hatches the King transferring the candidate into the spawned mob and initiates the next timer.
+/obj/effect/alien/resin/king_cocoon/proc/hatch_king()
+	icon_state = "hatched"
+	hatched = TRUE
+
+	QDEL_LIST(blockers)
+
+	var/mob/living/carbon/xenomorph/king/king = new(get_turf(src))
+	if(chosen_candidate?.mob)
+		var/mob/old_mob = chosen_candidate.mob
+		old_mob.mind.transfer_to(king)
+
+		if(isliving(old_mob) && old_mob.stat != DEAD)
+			old_mob.free_for_ghosts(TRUE)
+	else
+		king.free_for_ghosts(TRUE)
+	playsound(src, 'sound/voice/alien_queen_command.ogg', 75, 0)
+
+	chosen_candidate = null
+
+	// Gives some time for the King to get their barings before it can be OBed
+	addtimer(CALLBACK(src, PROC_REF(remove_ob_protection)), 1 MINUTES, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+
+/// The final step in the cocoon sequence: Resets pylon protection levels
+/obj/effect/alien/resin/king_cocoon/proc/remove_ob_protection()
+	var/datum/hive_status/hive = GLOB.hive_datum[hive_number]
+	for(var/obj/effect/alien/resin/special/pylon/pylon as anything in hive.active_endgame_pylons)
+		pylon.protection_level = initial(pylon.protection_level)
+		pylon.update_icon()
+
 /obj/item/explosive/grenade/alien
```

Comment:
```suggestion
/obj/effect/alien/resin/king_cocoon/attack_alien(mob/living/carbon/xenomorph/M)
    if(M.can_destroy_special() || hatched)
        return ..()
    return XENO_NO_DELAY_ACTION

/obj/item/explosive/grenade/alien
```

Otherwise you will still have lesser trolls destroying it.